service/src/main/java/com/epam/esm/dto/GiftCertificatePatchDto.java

```diff
@@ -0,0 +1,113 @@
+package com.epam.esm.dto;
+
+import javax.validation.Valid;
+import javax.validation.constraints.*;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * This DTO is validated to partially update a GiftCertificate entity.
+ *
+ * @since 1.0
+ */
+public class GiftCertificatePatchDto {
```

Comment:
Why did you created another dao for patch and did not use existing certificate dao?

Reply:
GiftCertificateDao is used for full entity update(put request), so it's validation doesn't let to pass null or empty value basic parameters(name, description, etc).
GiftCertificatePatchDao is used for partial update(patch request) and it's validation let passing null or empty value params.

Reply:
Just because of adding annotations @NotBlank you need to create another big dto and use it... Please think about another ways of validation to get rid of redundant duplicate dto

Reply:
Commit a6bc82a